src/ServiceControl.Audit.Persistence.PostgreSQL/PostgreSQLPersistenceInstaller.cs

```diff
@@ -0,0 +1,238 @@
+                time_sent
+            );", connection))
+        {
+            await cmd.ExecuteNonQueryAsync(cancellationToken);
+        }
+
+        await using (var cmd = new NpgsqlCommand(@"
+            CREATE INDEX IF NOT EXISTS idx_processed_messages_by_critical_time ON processed_messages (
+                critical_time
+            );", connection))
+        {
+            await cmd.ExecuteNonQueryAsync(cancellationToken);
+        }
+
+        await using (var cmd = new NpgsqlCommand(@"
+            CREATE INDEX IF NOT EXISTS idx_processed_messages_by_processing_time ON processed_messages (
+                processing_time
+            );", connection))
+        {
+            await cmd.ExecuteNonQueryAsync(cancellationToken);
+        }
+
+        await using (var cmd = new NpgsqlCommand(@"
+            CREATE INDEX IF NOT EXISTS idx_processed_messages_by_delivery_time ON processed_messages (
+                delivery_time
+            );", connection))
+        {
+            await cmd.ExecuteNonQueryAsync(cancellationToken);
+        }
+
+        await using (var cmd = new NpgsqlCommand(@"
+            CREATE INDEX IF NOT EXISTS idx_processed_messages_by_message_id ON processed_messages (
+                message_id
+            );", connection))
+        {
+            await cmd.ExecuteNonQueryAsync(cancellationToken);
+        }
+
+
+        await using (var cmd = new NpgsqlCommand(@"
+            CREATE INDEX IF NOT EXISTS idx_processed_messages_by_created_at ON processed_messages (
+                created_at
+            );", connection))
+        {
+            await cmd.ExecuteNonQueryAsync(cancellationToken);
+        }
+
+        await using (var cmd = new NpgsqlCommand(@"
+            CREATE INDEX IF NOT EXISTS idx_processed_messages_by_conversation ON processed_messages (
+                conversation_id,
+                time_sent
+            );", connection))
+        {
+            await cmd.ExecuteNonQueryAsync(cancellationToken);
+        }
+
+        await using (var cmd = new NpgsqlCommand(@"
+            CREATE INDEX IF NOT EXISTS idx_processed_messages_by_query ON processed_messages (
+                query
+            );", connection))
+        {
+            await cmd.ExecuteNonQueryAsync(cancellationToken);
+        }
+
+        await using (var cmd = new NpgsqlCommand(@"
+            CREATE INDEX IF NOT EXISTS idx_processed_messages_audit_counts ON processed_messages (
+                receiving_endpoint_name, processed_at
+            );", connection))
+        {
+            await cmd.ExecuteNonQueryAsync(cancellationToken);
+        }
+
+        // Create saga_snapshots table
+        await using (var cmd = new NpgsqlCommand(@"
+            CREATE TABLE IF NOT EXISTS saga_snapshots (
+                id UUID PRIMARY KEY,
+                saga_id UUID,
+                saga_type TEXT,
+                changes JSONB,
```

Comment:
Also, this design is based on the Raven design where we append changes to the changes column.
But for never ending sagas this could get quite large and in essence not work.
I think we need to keep rotating changes where we keep a maximum of x changes that is it. Maybe a bit smarter so that we always keep the first change, ...